operator/controllers/btpoperator_controller.go

```diff
@@ -403,9 +408,53 @@ func (r *BtpOperatorReconciler) SetupWithManager(mgr ctrl.Manager) error {
 			handler.EnqueueRequestsFromMapFunc(r.reconcileRequestForOldestBtpOperator),
 			builder.WithPredicates(r.watchSecretPredicates()),
 		).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(reconcileConfig),
+			builder.WithPredicates(r.watchConfigPredicates()),
+		).
 		Complete(r)
 }
 
+func reconcileConfig(object client.Object) []reconcile.Request {
+	logger := log.FromContext(nil, "name", object.GetName(), "namespace", object.GetNamespace())
+	cm, ok := object.(*corev1.ConfigMap)
+	if !ok {
+		return []reconcile.Request{}
+	}
+	logger.Info("reconciling config update", "config", cm.Data)
+	for k, v := range cm.Data {
+		var err error
+		switch k {
+		case "ChartPath":
+			ChartPath = v
+		case "ChartNamespace":
+			ChartNamespace = v
+		case "SecretName":
+			SecretName = v
+		case "ConfigName":
+			ConfigName = v
+		case "DeploymentName":
+			DeploymentName = v
+		case "ProcessingStateRequeueInterval":
+			ProcessingStateRequeueInterval, err = time.ParseDuration(v)
+		case "ReadyStateRequeueInterval":
+			ReadyStateRequeueInterval, err = time.ParseDuration(v)
+		case "ReadyTimeout":
+			ReadyTimeout, err = time.ParseDuration(v)
+		case "RetryInterval":
+			RetryInterval, err = time.ParseDuration(v)
+		default:
+			logger.Info("unknown config update key", k, v)
+		}
+		if err != nil {
+			logger.Info("failed to parse config update", k, err)
+		}
+	}
+
+	return []reconcile.Request{}
```

Comment:
@PK85 at the moment, it just changes configuration knobs, timeouts and intervals but it doesn't requeue any resources in the queue. Would you like to requeue all? It can cause difficult to estimate performance spikes on the controller.

Reply:
@wozniakjan what performance spikes do you talk about?

Reply:
if we desire to requeue all resources when any of the timeouts or requeue periods change via `ConfigMap`.